internal/security/bootstrapper/tcp/listener.go

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ * Copyright 2021 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ *******************************************************************************/
+
+package tcp
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+const (
+	connectionTimeout = 5 * time.Second
+)
+
+type TcpServer struct {
+	accepted bool
+	lock     *sync.Mutex
+}
+
+func NewTcpServer() *TcpServer {
+	return &TcpServer{lock: &sync.Mutex{}}
+}
+
+// StartListener will instantiate a new listener on port and optional host if it is not empty
+// returns error if failed to create a listener on the port number
+func (tcpSrv *TcpServer) StartListener(port int, lc logger.LoggingClient, host string) error {
+	lc.Debug(fmt.Sprintf("Starting listener on port %d ...", port))
+
+	trimmedHost := strings.TrimSpace(host)
+	doneSrv := fmt.Sprintf("%s:%d", trimmedHost, port)
+
+	listener, err := net.Listen("tcp", doneSrv)
+	if err != nil {
+		// nolint: staticcheck
+		return fmt.Errorf("Failed to create TCP listener: %v", err)
+	}
+
+	lc.Info(fmt.Sprintf("Security bootstrapper starts listening on tcp://%s", doneSrv))
+	for {
+		conn, err := listener.Accept()
+		if err != nil {
+			lc.Error(fmt.Sprintf("found error when accepting connection: %v ! Will retry again", err))
+			time.Sleep(time.Second)
+			continue
+		}
+
+		tcpSrv.lock.Lock()
+		tcpSrv.accepted = true
+		tcpSrv.lock.Unlock()
+
+		lc.Info(fmt.Sprintf("Accepted connection on port %d", port))
+
+		// once reached here, the connection is established, and consider the semaphore on this port is raised
+		go func(c *net.Conn) {
+			defer func() {
+				_ = (*c).Close()
+			}()
+
+			if err := handleConnection(*c, lc); err != nil {
+				lc.Warn(fmt.Sprintf("failed to write trough connection: %v", err))
+			}
+
+			// intended process listener is done
+			lc.Debug(fmt.Sprintf("connection on port %d is done", port))
```

Comment:
used new `Debugf` w/o the Sprintf. Same for all you logs that use Sprintf

Reply:
where is this new `Debugf` ? you meant `lc.Debugf()`?

Reply:
Yep.